torch/_dynamo/tensor_version_op.py

```diff
@@ -0,0 +1,57 @@
+import torch
+from torch._prims import _make_prim, RETURN_TYPE
+from torch._subclasses import FakeTensorMode
+from torch._subclasses.functional_tensor import FunctionalTensorMode
+
+_tensor_version = _make_prim(
+    schema="_tensor_version(Tensor self) -> SymInt",
+    return_type=RETURN_TYPE.NEW,
+    meta=torch.ops.aten._version.default,
+    impl_aten=torch.ops.aten._version.default,
+    doc="Tracable unbacked SymInt version of torch.Tensor._version",
+)
+
+
+@_tensor_version.py_impl(FakeTensorMode)
+def _tensor_version_fake(self):
+    """
+    The initial dynamo capture of _tensor_version + _unsafe_set_version_counter turns the
+    `._version` into an unbacked SymInt so that we don't need to specialize on the `._version`
+    of input tensors to the graph.
+    """
+    return self.fake_mode.shape_env.create_unbacked_symint()
```

Comment:
I don't really think this should be an unbacked symints. Unbacked SymInts indicate some sort of data dependent quantity, but that's not really what's going on here, you're just trying to generate a version int that's going to be erased later. In particular, an unbacked symint here is error prone if we do a deferred runtime assert later on it, this will result in an assert on the version count which will then get erased later (oops). What's so bad about just returning some hardcoded number here?

Reply:
The hard-coded number would become a ConstantVariable that dynamo would specialize on.  So it would require creating a new class in dynamo to handle the specially to prevent that.